### PR TITLE
Portable Memory Wiper.

### PR DESCRIPTION
**Pull request by alsoandanswer**
Hiya folks, Cave Johnson here.

I can't really think of a bit right now so just imagine reading everything in my voice for a few minutes.

https://user-images.githubusercontent.com/42310821/185928319-645124fe-439c-4a9b-9993-894079395714.mp4

### CODE REVIEW

**Comment by TheStryker** [comment]
Synth team approves.

File: code/game/objects/items/devices/memorywiper.dm [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		to_chat(user, SPAN_NOTICE("You initialize the memory wipe protocols. This procedure will take approximately 2 minutes."))
+		to_chat(attached, SPAN_WARNING("The computer hums to life and you feel your memories bleed away into nothingness."))
+		playsound(src.loc, /decl/sound_category/keyboard_sound, 30, TRUE)
+		wiping = TRUE
+		update_icon()
+		sleep(1200-rand(0,150))
+		if(attached && wiping)
+			visible_message(SPAN_NOTICE("\The [src] pings, \"Memory wipe protocols complete.\""))
+			playsound(src.loc, 'sound/machines/ping.ogg', 50, 0)
+			to_chat(attached, "<b>The process finishes, leaving you with nothing beyond your base programming and databases.</b>")
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/memorywiper.dm:72]: **Strongly** reccomend adding a ghost prompt here. Most people don't want to play non-characters and would rather just go do something else.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/memorywiper.dm [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	desc = "Inset into a sturdy pelican case, this computer holds the software and wiring necessary to wipe and factory reset any IPC."
+	desc_info = "You can alt-click the laptop while it's set down on surface to open it up and work with it. Left clicking while it is open will allow you to operate it."
+	icon = 'icons/obj/memorywiper.dmi'
+	icon_state = "portable_memorywiper"
+	item_state = "portable_memorywiper"
+	contained_sprite = TRUE
+	anchored = FALSE
+	drop_sound = 'sound/items/drop/backpack.ogg'
+	pickup_sound = 'sound/items/pickup/backpack.ogg'
+	var/mob/living/carbon/human/attached = null
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/memorywiper.dm:13]: Clean this up in Destroy().

File: code/game/objects/items/devices/memorywiper.dm [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+					visible_message("[usr] plugs \the [src] into \the [attached]'s maintenance port.")
+
+/obj/item/device/memorywiper/attack_hand(user as mob)
+	if(attached)
+		to_chat(user, SPAN_NOTICE("You initialize the memory wipe protocols. This procedure will take approximately a minute."))
+		to_chat(attached, SPAN_WARNING("The computer hums to life and you feel your memories bleed away into nothingness."))
+		playsound(src.loc, /decl/sound_category/keyboard_sound, 30, TRUE)
+		wiping = TRUE
+		update_icon()
+		sleep(600-rand(0,75))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/memorywiper.dm:68]: This shouldn't be a sleep, but a timer that leads to another proc. We shouldn't sleep in attack_hand as this sleeps the entire proc.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]
Put the wiper in the RD's locker and not on the table.

**Comment by alsoandanswer** [comment]
!review

**Comment by Snowy1237** [comment]
Does it work only on IPCs or on borgs too?

**Review by Arrow768 — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**